src/CMakeLists.txt

```diff
@@ -130,6 +130,8 @@ function(valhalla_module)
   set(libvalhalla_link_libraries "${libvalhalla_link_libraries};${external_module_depends}" CACHE INTERNAL "")
 endfunction()
 
+add_subdirectory(${VALHALLA_SOURCE_DIR}/third_party/robin-hood-hashing ${CMAKE_BINARY_DIR}/third_party/robin-hood-hashing)
```

Comment:
this makes me wonder why we dont include rapidjson this way. specifically what happens to downstream users of libvalhalla when they see this header in our headers? ideally we are also vendoring these headers inside of `/usr/local/include/valhalla/third_party` but im not sure if that happens automatically here or not. do you know @danpat?

Reply:
Depends on whether the header I've added this to is exposed publicly or not.

CMake has the `PUBLIC/PRIVATE` thing when you do `target_include_directories` - if one is marked as `PRIVATE`, then you're stating that a consumer of our CMake build as a sub-project won't need that.

Is `costmatrix.h` exposed in the public API interface?  If so, then yeah, doing this would mean `robin_hood.h` is needed by downstream builders.

Should be easy for this *particular* dependency, but I can see that kind of problem spiraling out of control.

What's the design pattern that's in place for defining the public Valhalla library interface?

Reply:
Yep, pretty much any third party include in `/valhalla/*/*.h` should be marked `PUBLIC`. I think we have at least 2 cases of these so far (date and rapidjson), but both of those dont use `add_subdirectory` they are manually specified. i'll have to clean that up at some point. but yeah this one, since it appears in one of our headers should be marked `PUBLIC` too